.github/workflows/build-and-test.yml

```diff
@@ -22,20 +25,32 @@ jobs:
         agent: [1, 2, 3]
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v3
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
 
       - name: Start Nx Agent ${{ matrix.agent }}
-        run: yarn nx-cloud start-agent
+        run: npx nx-cloud start-agent
 
   build:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v3
```

Comment:
why can't we put those two into the setup step as well? looks like it is duplicated

Reply:
We can't use a local action without checking out the source code first which is why we had duplicate `actions/checkout` steps. We could keep `nrwl/nx-set-shas` but it isn't required for jobs that don't use `nx affected` commands.